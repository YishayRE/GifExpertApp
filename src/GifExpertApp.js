import React, { Fragment, useState } from 'react'
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

const GifExpertApp = ({defaultCategory = []}) => {
//solo si jamas cambia
    //const categories = ['One Punch', 'Samurai X', 'Dragon Ball'];
    //const [categories, setCategories] = useState(['One Punch']);
    const [categories, setCategories] = useState(defaultCategory);

    //console.log(categories)

    return (
        <Fragment>
            <h2>GifExpertApp</h2>

            <AddCategory setCategories={setCategories}/>
            <hr />

            <ol>
                {
                    categories.map(category => 
                        <GifGrid 
                            key={category}
                            category={category}
                        />
                    )
            
                /*
                    categories.map(category => { <GifGrid />
                        return <li key={category}>{category}</li>;
                    })
                */
                }
            </ol>
        </Fragment>
    )
}

export default GifExpertApp;