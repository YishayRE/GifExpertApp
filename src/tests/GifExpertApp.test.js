import '@testing-library/jest-dom';
import React from "react";
import { shallow } from "enzyme";
import GifExpertApp from '../GifExpertApp';

describe('Pruebas en el componente GifExpertApp', () => {

    
    test('Debe de mostrar <GifExpertApp/>', () => {
        const wrapper = shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();
    });
    
    test('Debe de mostrar una lista de categorias', () => {
        const categories = ['One Punch', 'Dragon'];
        const wrapper = shallow(<GifExpertApp defaultCategory={categories}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
});