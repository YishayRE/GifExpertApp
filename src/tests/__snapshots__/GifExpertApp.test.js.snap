// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifExpertApp Debe de mostrar <GifExpertApp/> 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en el componente GifExpertApp Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon"
      key="Dragon"
    />
  </ol>
</Fragment>
`;
